ee/tables/execparsers/socketfilterfw/parser.go

```diff
@@ -0,0 +1,125 @@
+package socketfilterfw
+
+import (
+	"bufio"
+	"io"
+	"regexp"
+	"strings"
+)
+
+var appRegex = regexp.MustCompile("(.*)(?:\\s\\(state:\\s)([0-9]+)")
+var lineRegex = regexp.MustCompile("(state|block|built-in|downloaded|stealth|log mode|log option)(?:.*\\s)([0-9a-z]+)")
+
+// socketfilterfw returns lines for each `get` argument supplied.
+// The output data is in the same order as the supplied arguments.
+//
+// This supports parsing the list of apps and their allow state, or
+// each line describes a part of the feature and what state it's in.
+//
+// These are not very well-formed, so I'm doing some regex magic to
+// know which option the current line is, and then sanitize the state.
+func socketfilterfwParse(reader io.Reader) (any, error) {
+	results := make([]map[string]string, 0)
+	row := make(map[string]string)
+	parseAppData := false
+
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		// When parsing the app list, the first line of output is a total
+		// count of apps. We can break on this line to start parsing apps.
+		if strings.Contains(line, "Total number of apps") {
+			parseAppData = true
+			continue
+		}
+
+		if parseAppData {
+			appRow := parseAppList(line)
+			if appRow != nil {
+				results = append(results, appRow)
+			}
+
+			continue
+		}
```

Comment:
I think you could combine like this -- it would let you get rid of `parseAppData` bool entirely.

```suggestion
		if strings.Contains(line, "Total number of apps") {
			if appRow := parseAppList(line); appRow != nil {
				results = append(results, appRow)
			}
			continue
		}
```

Reply:
I'm confused how this is supposed to work. Only the first line of text for the app list contains that string:
```
Total number of apps = 6
replicatord (state: 1)
Pop Helper.app (state: 4)
Google Chrome (state: 4)
rtadvd (state: 1)
com.docker.backend (state: 1)
sshd-keygen-wrapper (state: 1)
```

Reply:
Oh blah, I'm sorry, I see what I missed! Ignore this.

Reply:
No worries, I just wasn't sure if I was missing something